src/cf-buttons/src/atoms/buttons-with-icons.less

```diff
@@ -1,1549 +1,793 @@
 @import '../atoms/buttons.less';
 
-//
-// Button icons
-//
-
 // Icon directions
 
 .a-btn_icon-position-left {
   &:before {
     padding-right: unit( 10.5px / @btn-font-size, em );
-    border-right: 1px solid @btn-text; // Fallback
-    border-right: 1px solid fade( @btn-text, 40% );
+    border-right: 1px solid mix( @btn-bg, @btn-text, 50% );;
     margin-right: unit( 7px / @btn-font-size, em );
   }
-
-  .a-btn__secondary &:before {
-    border-right-color: @btn__secondary-text; // Fallback
-    border-right-color: fade( @btn__secondary-text, 40% );
-  }
-
-  .a-btn__warning &:before {
-    border-right-color: @btn__warning-text; // Fallback
-    border-right-color: fade( @btn__warning-text, 40% );
-  }
-
-  .a-btn__disabled &:before,
-  .a-btn[disabled] &:before {
-    border-right-color: @btn__disabled-text; // Fallback
-    border-right-color: fade( @btn__disabled-text, 40% );
-  }
 }
 
 .a-btn_icon-position-right {
   &:after  {
-    padding-left: unit( 10.5px / @btn-font-size, em);
-    border-right: 0;
-    border-left: 1px solid @btn-text; // Fallback
-    border-left: 1px solid fade(@btn-text, 40%);
-    margin-left: unit( 7px / @btn-font-size, em);
+    padding-left: unit( 10.5px / @btn-font-size, em );
+    border-left: 1px solid mix( @btn-bg, @btn-text, 50% );;
+    margin-left: unit( 7px / @btn-font-size, em );
   }
-  .a-btn__secondary & {
-    border-left-color: @btn__secondary-text; // Fallback
-    border-left-color: fade(@btn__secondary-text, 40%);
-  }
-  .a-btn__warning & {
-    border-left-color: @btn__warning-text; // Fallback
-    border-left-color: fade(@btn__warning-text, 40%);
+}
+
+
+.a-btn_icon-position-left:before,
+.a-btn_icon-position-right:after {
+  font-family: 'CFPB Minicons';
+
+  .a-btn__secondary& {
+    border-color: mix( @btn__secondary-bg, @btn__secondary-text, 50% );
```

Comment:
This is new to me!

TIL for other folks out there.
http://lesscss.org/functions/#color-operations-mix